client/src/main/java/io/pravega/client/batch/impl/BatchClientFactoryImpl.java

```diff
@@ -174,4 +216,106 @@ public List<SegmentRange> getSegmentRangeBetweenStreamCuts(final StreamCut start
         return segmentRanges;
     }
 
+    @Override
+    public StreamCut getNextStreamCut(final StreamCut startingStreamCut, long approxDistanceToNextOffset) {
+        log.debug("getNextStreamCut() -> startingStreamCut = {}, approxDistanceToNextOffset = {}", startingStreamCut, approxDistanceToNextOffset);
+        Preconditions.checkArgument(approxDistanceToNextOffset > 0, "Ensure approxDistanceToNextOffset must be greater than 0");
+        Stream stream = startingStreamCut.asImpl().getStream();
+        Map<Segment, Long> nextPositionsMap = new HashMap<>();
+        Map<Segment, Long> scaledSegmentsMap = new HashMap<>();
+        for (Map.Entry<Segment, Long> positions : startingStreamCut.asImpl().getPositions().entrySet()) {
```

Comment:
Requirement here is to read X bytes from Stream(provided streamCuts). From the logic written here we would be ending up reading NxX bytes instead of X bytes.
According to logic, calculation for next offset ie X byte away from current offset from each segment. With this approach at the end it actually gives streamcuts to read NxX bytes data(if number of segments are 'N' and approxDistanceToNextOffset is 'X' bytes) which deviate from actual requirement.

Reply:
Updated the logic to fetch nextOffset for each segment based on requested approxDistanceToNextOffset per number of segments in the requested streamcut.